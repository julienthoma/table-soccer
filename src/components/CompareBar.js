import React from 'react';
import PropTypes from 'prop-types';
import './CompareBar.scss';

const CompareBar = ({
  leftHeadline,
  rightHeadline,
  factorLeft,
  leftValue,
  rightValue
}) => (
  <div styleName="root">
    <div styleName="headlineContainer">
      <h4 styleName="leftHeadline">{leftHeadline}</h4>
      <h4 styleName="rightHeadline">{rightHeadline}</h4>
    </div>

    <div styleName="bar">
      <div styleName="left" style={{ flexGrow: factorLeft }} />
      <div styleName="leftText">{leftValue}</div>
      <div styleName="right" />
      <div styleName="rightText">{rightValue}</div>
    </div>
  </div>
);

CompareBar.propTypes = {
  leftHeadline: PropTypes.string.isRequired,
  rightHeadline: PropTypes.string.isRequired,
  factorLeft: PropTypes.number.isRequired,
  leftValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  rightValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired
};

export default CompareBar;
