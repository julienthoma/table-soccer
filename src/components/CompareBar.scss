@import '../scss/colors.scss';

.root {
  width: 100%;
  padding: 5px 0;
}

.headlineContainer {
  display: flex;
  justify-content: space-between;
}

.leftHeadline, .rightHeadline {
  margin: 0;
}

.headlineLeft {
  margin: 0;
}

.headlineRight {
  margin: 0;
}

.bar {
  position: relative;
  display: flex;
  width: 100%;
}

.left {
  background: $blue;
  height: 36px;
}

.right {
  background: $pink;
  height: 36px;
  flex-grow: 1;
}

.leftText, .rightText {
  position: absolute;
  top: 10px;
  text-transform: uppercase;
  font-size: 12px;
}

.leftText {
  left: 5px;
}

.rightText {
  right: 5px;
}
