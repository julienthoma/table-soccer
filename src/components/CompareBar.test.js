import React from 'react';
import { renderWithMuiTheme } from '../test.utils';
import CompareBar from './CompareBar';

it('CompareBar renders correctly', () => {
  const bar = (
    <CompareBar
      leftHeadline="Hello"
      rightHeadline="World"
      leftValue={2}
      rightValue={3}
    />
  );
  expect(renderWithMuiTheme(bar)).toMatchSnapshot();
});
