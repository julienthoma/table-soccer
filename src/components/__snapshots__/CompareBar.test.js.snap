// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareBar renders correctly 1`] = `
.c2 {
  margin: 0;
  font-size: 15px;
  color: #6d6d6d;
  font-weight: 400;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  position: relative;
  width: 100%;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c5 {
  position: relative;
  background: #00bcd4;
  height: 20px;
  -webkit-box-flex: 0.4;
  -webkit-flex-grow: 0.4;
  -ms-flex-positive: 0.4;
  flex-grow: 0.4;
}

.c6 {
  position: relative;
  background: #ff4081;
  height: 20px;
  -webkit-box-flex: 0.6000000000000001;
  -webkit-flex-grow: 0.6000000000000001;
  -ms-flex-positive: 0.6000000000000001;
  flex-grow: 0.6000000000000001;
}

.c7 {
  position: absolute;
  top: 3px;
  text-transform: uppercase;
  font-size: 13px;
  left: 10px;
}

.c8 {
  position: absolute;
  top: 3px;
  text-transform: uppercase;
  font-size: 13px;
  right: 10px;
}

.c0 {
  width: 100%;
  padding: 5px 0;
  overflow: hidden;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h4
      className="c2"
    >
      Hello
    </h4>
    <h4
      className="c2"
    >
      World
    </h4>
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
    >
      <div
        className="c5"
        color="#00bcd4"
      />
      <div
        className="c6"
        color="#ff4081"
      />
    </div>
    <div
      className="c7"
    >
      2
    </div>
    <div
      className="c8"
    >
      3
    </div>
  </div>
</div>
`;
